### PR TITLE
[Account Abstraction] Message filter authenticator

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

Add an authenticators that allows users to filter specific messages

## Testing and Verifying

simple unit tests

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/authenticator/authenticator/message_filter_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		},
+
+		{"bank send. any match",
+			`{}`,
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("bar", 100))},
+			true,
+		},
+
+		{"bank send. using map as generic",
+			`{{"type":"/cosmos.bank.v1beta1.MsgSend","value":{}}`,
```
Comment (PaddyMc) [inline x/authenticator/authenticator/message_filter_test.go:111]: There's an extra `{` here, that should removed

File: x/authenticator/authenticator/message_filter_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			&bank.MsgSend{FromAddress: "from", ToAddress: "to", Amount: sdk.NewCoins(sdk.NewInt64Coin("bar", 100))},
+			true,
+		},
+
+		{"bank send. empty array as generic for arrays",
```
Comment (PaddyMc) [inline x/authenticator/authenticator/message_filter_test.go:116]: Would you mind adding add a test for the poolmanager swap amount in messages as well? Just cause that's probably the main use case?

**Review by PaddyMc — CHANGES_REQUESTED** [review]
Looks great so far!

Could you also register the authenticator in the manager?

File: x/authenticator/authenticator/message_filter_test.go (hunk 1 above)
Comment (nicolaslara) [reply x/authenticator/authenticator/message_filter_test.go:111]: hmm, interesting that that test passes 🤔

File: x/authenticator/authenticator/message_filter_test.go (hunk 2 above)
Comment (nicolaslara) [reply x/authenticator/authenticator/message_filter_test.go:116]: sure!

File: x/authenticator/authenticator/message_filter_test.go (hunk 1 above)
Comment (nicolaslara) [reply x/authenticator/authenticator/message_filter_test.go]: ahh, it didn't pass. Great! fixed now

File: x/authenticator/authenticator/message_filter_test.go (hunk 2 above)
Comment (nicolaslara) [reply x/authenticator/authenticator/message_filter_test.go:117]: added

**Review by PaddyMc — APPROVED** [review]
GG :rocket:

**Pull request merged**